--- Vulkan-Tools-1.4.349.orig/cube/CMakeLists.txt	2026-04-14 21:46:33.557525352 -0300
+++ Vulkan-Tools-1.4.349/cube/CMakeLists.txt	2026-04-14 21:46:46.581306079 -0300
@@ -67,7 +67,7 @@
     endif()
 endif()
 
-if(ANDROID OR APPLE)
+if(FALSE)
     set(WSI_DISPLAY_DEFAULT_SETTING "OFF")
 else()
     set(WSI_DISPLAY_DEFAULT_SETTING "ON")
@@ -175,7 +175,7 @@
 if(WIN32)
     add_definitions(-DWIN32_LEAN_AND_MEAN -DNOMINMAX)
     list(APPEND ENABLED_CUBE_PLATFORMS VK_USE_PLATFORM_WIN32_KHR)
-elseif(ANDROID)
+elseif(FALSE)
     list(APPEND ENABLED_CUBE_PLATFORMS VK_USE_PLATFORM_ANDROID_KHR)
 elseif(APPLE)
     list(APPEND ENABLED_CUBE_PLATFORMS VK_USE_PLATFORM_METAL_EXT)
@@ -293,25 +293,7 @@
 target_include_directories(vkcube PRIVATE .)
 target_link_libraries(vkcube ${CMAKE_DL_LIBS} Vulkan::Headers)
 
-if (ANDROID)
-    install(TARGETS vkcube DESTINATION ${CMAKE_INSTALL_LIBDIR})
-elseif(APPLE)
-    install(
-        TARGETS vkcube
-        # Install runtime dependencies like the Vulkan::Loader so the app is self-contained
-        RUNTIME_DEPENDENCIES DESTINATION "cube/vkcube.app/Contents/Frameworks/"
-        BUNDLE DESTINATION "cube"
-    )
-    set_target_properties(vkcube PROPERTIES
-        INSTALL_RPATH @executable_path/../Frameworks
-    )
-else()
-    install(TARGETS vkcube)
-endif()
 
-if (ANDROID)
-    return()
-endif()
 
 # ----------------------------------------------------------------------------
 # vkcubepp
