--- a/cube/cube.c
+++ b/cube/cube.c
@@ -3914,7 +3914,11 @@ static void demo_check_and_set_wsi_platform(struct demo *demo) {
         }
     }
 #endif
 #if defined(VK_USE_PLATFORM_DISPLAY_KHR)
+#ifdef __ANDROID__
+    if (demo->wsi_platform == WSI_PLATFORM_DISPLAY) {
+#else
     if (demo->wsi_platform == WSI_PLATFORM_DISPLAY || demo->wsi_platform == WSI_PLATFORM_AUTO) {
+#endif
         bool display_extension_available = false;
         for (uint32_t i = 0; i < demo->enabled_extension_count; i++) {
             if (strcmp(demo->extension_names[i], VK_KHR_DISPLAY_EXTENSION_NAME) == 0) {
--- a/cube/cube.cpp
+++ b/cube/cube.cpp
@@ -1434,7 +1434,11 @@ void Demo::check_and_set_wsi_platform() {
         }
     }
 #endif
 #if defined(VK_USE_PLATFORM_DISPLAY_KHR)
+#ifdef __ANDROID__
+    if (wsi_platform == WsiPlatform::display) {
+#else
     if (wsi_platform == WsiPlatform::display || wsi_platform == WsiPlatform::auto_) {
+#endif
         auto found = std::find_if(enabled_instance_extensions.begin(), enabled_instance_extensions.end(),
                                   [](const char *str) { return 0 == strcmp(str, VK_KHR_DISPLAY_EXTENSION_NAME); });
         if (found != enabled_instance_extensions.end()) {
