diff --git a/dlls/dinput/joystick_micewine.c b/dlls/dinput/joystick_micewine.c
index 7935828f346..6ae115ad140 100644
--- a/dlls/dinput/joystick_micewine.c
+++ b/dlls/dinput/joystick_micewine.c
@@ -32,6 +32,7 @@
 #include "wine/debug.h"
 #include "wine/hid.h"
 
+
 #define CONTROLLER_BUFFER_SIZE 11
 #define BUFFER_SIZE (CONTROLLER_BUFFER_SIZE * 4)
 #define SERVER_PORT 7941
@@ -54,7 +55,7 @@
 WINE_DEFAULT_DEBUG_CHANNEL(dinput);
 
 DEFINE_GUID( micewine_joystick_guid, 0xdeadbeef, 0x7734, 0x11d2, 0x8d, 0x4a, 0x23, 0x90, 0x3f, 0xb6, 0xbd, 0xf7 );
-DEFINE_GUID (GUID_DEVINTERFACE_HID, 0x4D1E55B2L, 0xF16F, 0x11CF, 0x88, 0xCB, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30);
+DEFINE_GUID( GUID_DEVINTERFACE_HID, 0x4D1E55B2L, 0xF16F, 0x11CF, 0x88, 0xCB, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30);
 
 static bool dinput_is_ready = false;
 
@@ -243,7 +244,7 @@ static HRESULT micewine_joystick_get_property( IDirectInputDevice8W *iface, DWOR
         case (DWORD_PTR)DIPROP_GUIDANDPATH:
         {
             DIPROPGUIDANDPATH *value = (DIPROPGUIDANDPATH *)header;
-            value->guidClass = GUID_DEVCLASS_HIDCLASS;
+            value->guidClass = GUID_DEVINTERFACE_HID;
             lstrcpynW( value->wszPath, impl->device_path, MAX_PATH );
             return DI_OK;
         }
@@ -575,8 +576,8 @@ HRESULT micewine_joystick_enum_device( DWORD type, DWORD flags, DIDEVICEINSTANCE
     instance->wUsagePage = 0x01;
     instance->wUsage = 0x05;
 
-    swprintf(instance->tszInstanceName, MAX_PATH, L"MiceWine Virtual Controller %i", index);
-    wcscpy(instance->tszProductName,  L"MiceWine Product");
+    swprintf(instance->tszInstanceName, MAX_PATH, L"Xbox Controller %i", index);
+    wcscpy(instance->tszProductName,  L"Xbox 360 Controller");
 
 	return DI_OK;
 }
@@ -617,9 +618,9 @@ HRESULT micewine_joystick_create_device(struct dinput *dinput, const GUID *guid,
     impl->base.dwCoopLevel = DISCL_NONEXCLUSIVE | DISCL_BACKGROUND;
     impl->base.read_event = CreateEventW(NULL, FALSE, FALSE, NULL);
 
-    swprintf(impl->base.instance.tszInstanceName, MAX_PATH, L"MiceWine Virtual Controller %i", index);
-    wcscpy(impl->base.instance.tszProductName,  L"MiceWine Product");
-    swprintf(impl->device_path, MAX_PATH, L"\\\\?\\hid#vid_2563&pid_0575&ig_0%i#273&03006316632500007505000011010000.0&0&0&1#{4d1e55b2-f16f-11cf-88cb-001111000030}", index);
+    swprintf(impl->base.instance.tszInstanceName, MAX_PATH, L"Xbox Controller %i", index);
+    wcscpy(impl->base.instance.tszProductName,  L"Xbox 360 Controller");
+    swprintf(impl->device_path, MAX_PATH, L"\\\\?\\hid#vid_045e&pid_028e&ig_00%i#{4d1e55b2-f16f-11cf-88cb-001111000030}", index);
 
     impl->index = index;
     index++;
diff --git a/dlls/windows.gaming.input/main.c b/dlls/windows.gaming.input/main.c
index 39c0540aace..2154e50c829 100644
--- a/dlls/windows.gaming.input/main.c
+++ b/dlls/windows.gaming.input/main.c
@@ -40,7 +40,6 @@ WINE_DEFAULT_DEBUG_CHANNEL(input);
 
 HINSTANCE windows_gaming_input;
 
-DEFINE_GUID( GUID_DEVINTERFACE_WINEXINPUT,0x6c53d5fd,0x6480,0x440f,0xb6,0x18,0x47,0x67,0x50,0xc5,0xe1,0xa6 );
 
 static LRESULT CALLBACK devnotify_wndproc( HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam )
 {
@@ -72,19 +71,12 @@ static void initialize_providers( void )
     char buffer[offsetof( SP_DEVICE_INTERFACE_DETAIL_DATA_W, DevicePath[MAX_PATH] )];
     SP_DEVICE_INTERFACE_DETAIL_DATA_W *detail = (void *)buffer;
     SP_DEVICE_INTERFACE_DATA iface = {sizeof(iface)};
-    GUID guid = GUID_DEVINTERFACE_WINEXINPUT;
+    GUID guid;
     HDEVINFO set;
     DWORD i = 0;
 
     set = SetupDiGetClassDevsW( NULL, NULL, NULL, DIGCF_ALLCLASSES | DIGCF_DEVICEINTERFACE | DIGCF_PRESENT );
 
-    while (SetupDiEnumDeviceInterfaces( set, NULL, &guid, i++, &iface ))
-    {
-        detail->cbSize = sizeof(*detail);
-        if (!SetupDiGetDeviceInterfaceDetailW( set, &iface, detail, sizeof(buffer), NULL, NULL )) continue;
-        provider_create( detail->DevicePath );
-    }
-
     HidD_GetHidGuid( &guid );
 
     while (SetupDiEnumDeviceInterfaces( set, NULL, &guid, i++, &iface ))
