diff -uNr 58-android-shmem/shmem.c 58-android-shmem.mod/shmem.c
--- 58-android-shmem/shmem.c	2022-03-27 13:15:50.000000000 -0300
+++ 58-android-shmem.mod/shmem.c	2024-04-13 19:22:22.244764249 -0300
@@ -16,8 +16,12 @@
 
 #include "shm.h"
 
+#include <fcntl.h>
+#include <string.h>
+
 #define DBG(...) __android_log_print(ANDROID_LOG_INFO, "shmem", __VA_ARGS__)
 #define ASHV_KEY_SYMLINK_PATH _PATH_TMP "ashv_key_%d"
+#define _PATH_TMP "/data/data/com.micewine.emu/files/usr/tmp"
 #define ANDROID_SHMEM_SOCKNAME "/dev/shm/%08x"
 #define ROUND_UP(N, S) ((((N) + (S) - 1) / (S)) * (S))
 
@@ -399,6 +403,7 @@
 
 	shmem = realloc(shmem, shmem_amount * sizeof(shmem_t));
 	size = ROUND_UP(size, getpagesize());
+	if (size < 8 * 1024 * 1024) size = 8 * 1024 * 1024;
 	shmem[idx].size = size;
 	shmem[idx].descriptor = ashmem_create_region(buf, size);
 	shmem[idx].addr = NULL;
@@ -468,6 +473,11 @@
 			shmem[idx].addr = NULL;
 		}
 	}
+
+#ifdef MADV_HUGEPAGE
+	if (shmem[idx].addr)
+		madvise(shmem[idx].addr, shmem[idx].size, MADV_HUGEPAGE);
+#endif
+
 	addr = shmem[idx].addr;
 	DBG ("%s: mapped addr %p for FD %d ID %d", __PRETTY_FUNCTION__, addr, shmem[idx].descriptor, idx);
 	pthread_mutex_unlock (&mutex);
